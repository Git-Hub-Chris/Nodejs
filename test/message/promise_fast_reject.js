'use strict';

// Currently we keep a reference of unhandled rejections in JS and that prevents
// the GC to collect these before the process exits naturally.

const common = require('../common');
// TODO: Remove when making this the default.
process.env.NODE_UNHANDLED_REJECTION = 'ERROR_ON_GC';

new Promise(function(res, rej) {
  throw new Error('One');
});

Promise.reject(new Error('Two'));

process.on('uncaughtException', common.mustNotCall());

process.on('exit', (code) => process._rawDebug(`Exit code ${code}`));
