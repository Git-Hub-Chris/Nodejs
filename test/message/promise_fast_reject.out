(node:*) UnhandledPromiseRejectionWarning: Error: One
    at *promise_fast_reject.js:*
    at new Promise (<anonymous>)
    at Object.<anonymous> (*promise_fast_reject.js:*)
    at *
    at *
    at *
    at *
    at *
    at *
    at *
(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:*) UnhandledPromiseRejectionWarning: Error: Two
    at *promise_fast_reject.js:*
    at *
    at *
    at *
    at *
    at *
    at *
    at *
    at *
(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
Exit code 0
